datashader/tests/test_datatypes.py

```diff
@@ -835,6 +835,12 @@ def test_sort_values_frame(self):
     def test_where_series(self):
         pass
 
+    @pytest.mark.xfail(reason="not currently supported")
```

Comment:
Should we instead pin to pandas !=2.2, expecting them to fix it in the next release?

Reply:
The line you have commented on is a new test for ExtensionArray, which was added in Pandas 2.2.

I have updated the skip for the upstream regression.

Reply:
I'm confused, since it says `@pytest.mark.xfail(reason="not currently supported")` ("currently"), but in any case, sounds like you're on it!

Reply:
`duplicated` has likely never worked for `RaggedArray`; I have added a NotImplementedError for it in the class. Before that, this would give a wrong result. 
![image](https://github.com/holoviz/datashader/assets/19758978/f202477a-294c-4319-af9b-918265eefbd2)

``` python
import pandas as pd
from datashader.datatypes import RaggedArray

data = RaggedArray([[0, 1], [1, 2, 3, 4], [], [-1, -2], []], dtype='float64')
arr = data.take([0, 1, 0, 1])
arr.duplicated('first')

arr = pd.array([0, 1, 0, 1])
arr.duplicated('first')
```

Pandas use a hashtable to check if something is duplicated, but it is not designed to work with RaggedArray (which makes sense). I don't think it is impossible to support this, which is why I have added "currently". I hope that clarifies things.

Reply:
Ok, thanks! Maybe the way to clarify it would be to state _what_ doesn't currently support it, e.g. "not currently supported by Pandas", "not currently supported by RaggedArray", or "not currently supported by Datshader".